docs/development/go/developing-udp-and-tcp-clients-and-servers-in-go/index.md

```diff
@@ -0,0 +1,579 @@
+create TCP and UDP clients, respectively.
+
+The TCP part of this guide will use the generic functions whereas the UDP part will use
+`net.ResolveUDPAddr()`, `net.ListenUDP()` and `net.DialUDP()`.
+
+{{< note >}}
+All versions of `net.Dial()` and `net.Listen()` return data types that implement the
+`io.Reader` and `io.Writer` interfaces. This means that you can use regular File I/O
+functions to send and receive data from a TCP/IP connection if you want.
+{{< /note >}}
+
+{{< note >}}
+If you ever want to test a TCP/IP application, either a server or a client, you might
+find the `nc(1)` command line utility very handy.
+{{< /note >}}
+
+## A TCP Client
+
+In this section of the guide, we are going to see how to develop a generic TCP client in Go.
+The utility will allow you to interact with any TCP server.
+
+### Looking at the Go code of the TCP client
+
+The Go code of the TCP client is the following:
+
+{{< file "./tcpC.go" go >}}
+package main
+
+import (
+        "bufio"
+        "fmt"
+        "net"
+        "os"
+        "strings"
+)
+
+func main() {
+        arguments := os.Args
+        if len(arguments) == 1 {
+                fmt.Println("Please provide host:port.")
+                return
+        }
+
+        CONNECT := arguments[1]
+        c, err := net.Dial("tcp", CONNECT)
+        if err != nil {
+                fmt.Println(err)
+                return
+        }
+
+        for {
+                reader := bufio.NewReader(os.Stdin)
+                fmt.Print(">> ")
+                text, _ := reader.ReadString('\n')
+                fmt.Fprintf(c, text+"\n")
+
+                message, _ := bufio.NewReader(c).ReadString('\n')
+                fmt.Print("->: " + message)
+                if strings.TrimSpace(string(text)) == "STOP" {
+                        fmt.Println("TCP client exiting...")
+                        return
+                }
+        }
+}
+{{< /file >}}
+
+ - The first part of the `main()` function gathers command line arguments and makes sure that `host:port` was sent. This argument is then put into the `CONNECT` variable to be used in the `net.Dial()` call. The implementation of the TCP client begins
+with a call to `net.Dial()` that allows you to connect to the desired TCP server.
+The second parameter of `net.Dial()` has two virtual parts. The first part is the hostname
+or the IP address of the TCP server and the second part is the port number the TCP server
+listens to.
+
+ - The program reads user input using `bufio.NewReader(os.Stdin)` and `ReadString()`, which
+is sent to the TCP server over the network using `Fprintf()`.
+
+ - Last, the TCP client reads the response of the TCP server using another `bufio` reader and
+the `bufio.NewReader(c).ReadString('\n')` statement. The `error` variable is ignored here for
+reasons of simplicity only.
+
+ - This is an endless `for` loop that will only terminate when you send the word `STOP` to the TCP server.
+
```

Comment:
I also made the points under each file explaining code into bullet points for clarity and ease of reading.